### PR TITLE
Logging added to google cloud exporter

### PR DESCRIPTION
**Pull request by StefanKurek**
**Description:**
Added logging capabilities to the google cloud exporter. New config option log.name_fields were added to point to which OTEL attribute should be used for a name. Monitored resource for logs is always the global resource as there isn't an obvious way to determine other resource types

**Link to tracking Issue:** 
[#8249](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8249)

**Testing:**
Unit tests were added for any new files, and new unit tests were added to existing files that were modified

**Documentation:** 
Added to the README for any modified features.

### CODE REVIEW

**Comment by dashpole** [comment]
cc @damemi

**Review by damemi — COMMENTED** [review]
Thanks for opening this so we can check it out @StefanKurek. Would you be interested in creating a new PR in https://github.com/GoogleCloudPlatform/opentelemetry-operations-go to pull out the relevant implementation bits from here that will live in that repo?

File: exporter/googlecloudexporter/README.md [hunk 1]
```diff
@@ -60,6 +60,10 @@ These instructions are to get you up and running quickly with the GCP exporter i
           receivers: [otlp]
           processors: [memory_limiter, batch]
           exporters: [googlecloud, logging]
+        logs:
+          receivers: [otlp]
+          processors: [memeory_limiter, batch]
+          exporters: [googlecloud, logging]
     ```
 
 3.  **Set up credentials.**
```
[the remark this answers is not in the document]
Comment (dashpole) [reply exporter/googlecloudexporter/README.md:69]: Here is google cloud's logging IAM docs: https://cloud.google.com/logging/docs/access-control.  We probably want `roles/logging.logWriter`

**Comment by StefanKurek** [comment]
> Thanks for opening this so we can check it out @StefanKurek. Would you be interested in creating a new PR in https://github.com/GoogleCloudPlatform/opentelemetry-operations-go to pull out the relevant implementation bits from here that will live in that repo?

@damemi I just wanted to ping you to let you know that I'm not ignoring this request. I'm figuring out if I have the time to go this route.

**Comment by damemi** [comment]
@StefanKurek no problem, if you don't have time I have started some work on this myself. 

I talked to @dashpole about it and he suggested one option is we could merge your code to a separate branch (so it has the appropriate copyright release), and we can pick your logic from there. No pressure if you're busy, but we would like to make some progress without stepping on any toes.

**Comment by dashpole** [comment]
This was contributed to https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/tree/tmp_stanza_logging, and will be integrated into the logging exporter we are working on

**Pull request closed without merging**